### ISSUE TITLE
drand-client should not take a local port

### ISSUE THREAD

**Issue by hsanjuan** [drand/drand#719]
The client should not bind the libp2p host to a local port and it should not take a `port` parameter, as it is a client, not a relay to which other people will connect to.

**Comment by willscott** [comment]
It runs a libp2p host, You can not provide a port in which case it shouldn't listen. I guess you can end up with better libp2p connectivity if you do allow other libp2p nodes to connect to you?

**Comment by hsanjuan** [comment]
You can launch a libp2p host without listeners.

It is a client, it does not want other nodes to connect to it, for that the gossip-relay should be used, it certainly should not listen on 0.0.0.0 by default and I hope it is not trying to upnp its way through firewalls...

**Comment by willscott** [comment]
If no "--port" flag is specified, the client will not listen at all. (the
host will be created with `libp2p.NoListenAddrs`)

**Comment by hsanjuan** [comment]
Correct, but I don't understand why that flag exists. It has consequences that are not explained to the user, and the help text is wrong (it says it's `when connecting to relays` and it is actually about when relays connect to you (and you potentially become one?)). Does it make the client become a relay itself? I just don't find it appropriate that a client becomes a relay under the hood.

**Comment by kenyiu** [comment]
the `drand-client` is no longer in the codebase since v2, so no longer relevant.
